File: c++/217/source.cpp
/**
 * @author piaoyidage
 * @version 创建时间：2015-6
 *
 */	

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) 
    {
        set<int> s;
		for (vector<int>::iterator i = nums.begin(); i != nums.end(); ++i)
		{
			s.insert(*i);
		}

		if (s.size() != nums.size())
		{
			return true;
		}

		return false;
    }
};

File: java/14/source.java
/**
 * @author piaoyidage
 * @version 创建时间：2015-06-09
 *
 */ 


public class Solution {
   public String longestCommonPrefix(String[] strs)
    {
        String str = "";
        if (strs == null)
        {
            return str;
        }
        int count = 0;
        String cur = null;
        // 取首个字符
        if ((strs.length > 0) && (strs[0].length() > count))
            cur = strs[0].substring(count, count+1);
        else
            return str;
        // 逐个比较
        while (true)
        {
            for (String s : strs)
            {
                if ( (s.length() <= count) || (!s.substring(count, count+1).equals(cur)) )
                {
                    return str;
                }
            }
            str += cur;
            count++;
            if ((strs[0].length() > count))
                cur = strs[0].substring(count, count+1);
            else
                return str;
        }
    }
}
